x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java

```diff
@@ -469,6 +472,26 @@ private void authorizeAction(
                 l.onResponse(result);
             }));
         } else if (isIndexAction(action)) {
+            if (request instanceof SearchRequest searchRequest && searchRequest.pointInTimeBuilder() != null) {
+                /**
+                 * We override the {@link SearchRequest#indices} with the PIT's indices for the authorization check.
+                 * Although this copy also occurs in the rest layer, we cannot rely on it, as the action might be executed
+                 * directly in the transport client.
+                 * For backward compatibility reasons, we refrain from failing the request if {@link SearchRequest#indices}
+                 * is non-empty. Instead, we verify that the indices match the PIT definition and deny access if they don't.
+                 */
+                var indices = SearchContextId.decodeIndices(searchRequest.pointInTimeBuilder().getEncodedId());
+                if (searchRequest.indices().length > 0 && Arrays.equals(searchRequest.indices(), indices) == false) {
+                    IllegalArgumentException cause = new IllegalArgumentException(
+                        "indices defined in the [pit] don't match the request's indices"
+                    );
+                    auditTrail.accessDenied(requestId, authentication, action, request, authzInfo);
+                    listener.onFailure(actionDenied(authentication, authzInfo, action, request, cause));
+                    return;
+                }
+                searchRequest.indices(indices);
```

Comment:
do we still need to replace the indices or could we just perform the authorization and leave the indices empty? I am thinking that it could be good to enforce that a search request with a pit never specifies indices, but this would break that assumption.

Reply:
++, we don't need to replace the indices I agree. I pushed [89853df](https://github.com/elastic/elasticsearch/pull/103152/commits/89853df0447c35e3cc5620f6edd82de1f05c809e) to resolve the indices directly from the PIT. I also removed the copy in the rest action (this is not needed now since the security layer knows where to look) and moved the request validation to the transport implementation

Reply:
I was going to say, you may get some test failures from places that do set the indices. I think for instance ML transform does that.